src/presentation/http/router/note.test.ts

```diff
@@ -576,7 +655,47 @@ describe('Note API', () => {
   });
 
   describe('POST /note', () => {
-    test('Should correctly save relation to parent note if parentId passed', async () => {
+    test.each([
+      /**
+       * Specified extra tools
+       */
+      {
+        noteTools: [
+          {
+            name : headerTool.name,
+            id : headerTool.id,
+          },
+          {
+            name : listTool.name,
+            id : listTool.id,
+          },
+        ],
+        expectedStatusCode: 400,
+        expectedMessage: 'Incorrect tools passed',
+      },
+      /**
+       * All tools specified correctly
+       */
+      {
+        noteTools: [
+          {
+            name: headerTool.name,
+            id: headerTool.id,
+          },
+        ],
+        expectedStatusCode: 200,
+        expectedMessage: null,
+      },
+      /**
+       * Specified less tools
+       */
+      {
+        noteTools: [],
```

Comment:
cover all cases of validation please

Reply:
added cases for incorrect tool names and ids